The job **`i686-msvc-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/runs/5241931490?check_suite_focus=true) [(plain)](https://github.com/rust-lang-ci/rust/commit/493573589909b6e4e8463e65d1787654171d3b07/checks/5241931490/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [rustdoc] rustdoc\intra-doc\extern-lang-item-impl.rs stdout ----

error: auxiliary build of "D:\\a\\rust\\rust\\src/test\\rustdoc\\intra-doc\\auxiliary\\extern-lang-item-impl-dep.rs" failed to compile: 
status: exit code: 1
command: PATH="D:\a\rust\rust\build\i686-pc-windows-msvc\stage2\bin;C:\Program Files (x86)\Windows Kits\10\bin\x64;C:\Program Files (x86)\Windows Kits\10\bin\10.0.22000.0\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x86;D:\a\rust\rust\build\i686-pc-windows-msvc\stage0-bootstrap-tools\i686-pc-windows-msvc\release\deps;D:\a\rust\rust\build\i686-pc-windows-msvc\stage0\bin;D:\a\rust\rust\ninja;D:\a\rust\rust\msys2\mingw32\bin;C:\hostedtoolcache\windows\Python\3.10.1\x64\Scripts;C:\hostedtoolcache\windows\Python\3.10.1\x64;C:\msys64\usr\bin;D:\a\rust\rust\sccache;C:\Program Files\MongoDB\Server\5.0\bin;C:\aliyun-cli;C:\vcpkg;C:\cf-cli;C:\Program Files (x86)\NSIS;C:\tools\zstd;C:\Program Files\Mercurial;C:\hostedtoolcache\windows\stack\2.7.3\x64;C:\cabal\bin;C:\ghcup\bin;C:\tools\ghc-9.2.1\bin;C:\Program Files\dotnet;C:\mysql\bin;C:\Program Files\R\R-4.1.2\bin\x64;C:\SeleniumWebDrivers\GeckoDriver;C:\Program Files (x86)\sbt\bin;C:\Program Files (x86)\GitHub CLI;C:\Program Files\Git\bin;C:\Program Files (x86)\pipx_bin;C:\hostedtoolcache\windows\go\1.15.15\x64\bin;C:\hostedtoolcache\windows\Python\3.7.9\x64\Scripts;C:\hostedtoolcache\windows\Python\3.7.9\x64;C:\hostedtoolcache\windows\Ruby\2.5.9\x64\bin;C:\tools\kotlinc\bin;C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\8.0.322-6\x64\bin;C:\npm\prefix;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\ProgramData\kind;C:\Program Files\Eclipse Foundation\jdk-8.0.302.8-hotspot\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\ProgramData\Chocolatey\bin;C:\Program Files\Docker;C:\Program Files\PowerShell\7;C:\Program Files\Microsoft\Web Platform Installer;C:\Program Files\dotnet;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn;C:\Program Files\nodejs;C:\Program Files\OpenSSL\bin;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin;C:\Program Files\TortoiseSVN\bin;C:\Program Files\CMake\bin;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.8.4\bin;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\tools\php;C:\Program Files (x86)\sbt\bin;C:\SeleniumWebDrivers\ChromeDriver;C:\SeleniumWebDrivers\EdgeDriver;C:\Program Files\Amazon\AWSCLIV2;C:\Program Files\Amazon\SessionManagerPlugin\bin;C:\Program Files\Amazon\AWSSAMCLI\bin;C:\Program Files (x86)\Google\Cloud SDK\google-cloud-sdk\bin;C:\Program Files (x86)\Microsoft BizTalk Server;C:\Program Files\LLVM\bin;C:\Users\runneradmin\.dotnet\tools;C:\Users\runneradmin\.cargo\bin;C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps" "D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\stage2\\bin\\rustc.exe" "D:\\a\\rust\\rust\\src/test\\rustdoc\\intra-doc\\auxiliary\\extern-lang-item-impl-dep.rs" "-Zthreads=1" "--target=i686-pc-windows-msvc" "-C" "prefer-dynamic" "--out-dir" "D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\test\\rustdoc\\intra-doc\\extern-lang-item-impl\\auxiliary" "-Crpath" "-O" "-Cdebuginfo=0" "-Lnative=D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\native\\rust-test-helpers" "--crate-type" "dylib" "-L" "D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\test\\rustdoc\\intra-doc\\extern-lang-item-impl\\auxiliary"
------------------------------------------

------------------------------------------
stderr:
stderr:
------------------------------------------
error: linking with `link.exe` failed: exit code: 1120
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x86\\link.exe" "/DEF:C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcSZsGIZ\\lib.def" "/NOLOGO" "/LARGEADDRESSAWARE" "/SAFESEH" "D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\test\\rustdoc\\intra-doc\\extern-lang-item-impl\\auxiliary\\extern-lang-item-impl-dep.extern_lang_item_impl_dep.795515df-cgu.0.rcgu.o" "D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\test\\rustdoc\\intra-doc\\extern-lang-item-impl\\auxiliary\\extern-lang-item-impl-dep.extern_lang_item_impl_dep.795515df-cgu.1.rcgu.o" "D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\test\\rustdoc\\intra-doc\\extern-lang-item-impl\\auxiliary\\extern-lang-item-impl-dep.4wz85i668g7nf2em.rcgu.rmeta" "/LIBPATH:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\native\\rust-test-helpers" "/LIBPATH:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\test\\rustdoc\\intra-doc\\extern-lang-item-impl\\auxiliary" "/LIBPATH:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\stage2\\lib\\rustlib\\i686-pc-windows-msvc\\lib" "/WHOLEARCHIVE:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\stage2\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\librustc_std_workspace_core-55b3fa318a6249cc.rlib" "/WHOLEARCHIVE:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\stage2\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcore-5048a7bc95ecbe8e.rlib" "D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\stage2\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcompiler_builtins-fe791603c7d507a8.rlib" "/NXCOMPAT" "/LIBPATH:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\stage2\\lib\\rustlib\\i686-pc-windows-msvc\\lib" "/OUT:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\test\\rustdoc\\intra-doc\\extern-lang-item-impl\\auxiliary\\extern_lang_item_impl_dep.dll" "/OPT:REF,ICF" "/DLL" "/IMPLIB:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\test\\rustdoc\\intra-doc\\extern-lang-item-impl\\auxiliary\\extern_lang_item_impl_dep.dll.lib" "/DEBUG" "/NATVIS:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\stage2\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\stage2\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\stage2\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:D:\\a\\rust\\rust\\build\\i686-pc-windows-msvc\\stage2\\lib\\rustlib\\etc\\libstd.natvis"
  = note:    Creating library D:\a\rust\rust\build\i686-pc-windows-msvc\test\rustdoc\intra-doc\extern-lang-item-impl\auxiliary\extern_lang_item_impl_dep.dll.lib and object D:\a\rust\rust\build\i686-pc-windows-msvc\test\rustdoc\intra-doc\extern-lang-item-impl\auxiliary\extern_lang_item_impl_dep.dll.exp
          libcore-5048a7bc95ecbe8e.rlib(core-5048a7bc95ecbe8e.core.8a217689-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol _memset referenced in function __ZN79_$LT$core..num..dec2flt..decimal..Decimal$u20$as$u20$core..default..Default$GT$7default17h3e40ba340f81545eE
          libcore-5048a7bc95ecbe8e.rlib(core-5048a7bc95ecbe8e.core.8a217689-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol __aulldiv referenced in function __ZN4core3num7dec2flt7decimal7Decimal10left_shift17h741b4e38f6d4d572E
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.2.rcgu.o) : error LNK2001: unresolved external symbol __aulldiv
          libcore-5048a7bc95ecbe8e.rlib(core-5048a7bc95ecbe8e.core.8a217689-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol _memcpy referenced in function __ZN4core3num7flt2dec8strategy6dragon9mul_pow1017hd41edd8f878af714E
          libcore-5048a7bc95ecbe8e.rlib(core-5048a7bc95ecbe8e.core.8a217689-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol __aullrem referenced in function __ZN4core3fmt3num14parse_u64_into17hab38d7379529e387E
          libcore-5048a7bc95ecbe8e.rlib(core-5048a7bc95ecbe8e.core.8a217689-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol _memcmp referenced in function __ZN4core3str7pattern11StrSearcher3new17hc85eebf388869f10E
          libcore-5048a7bc95ecbe8e.rlib(core-5048a7bc95ecbe8e.core.8a217689-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol ___CxxFrameHandler3 referenced in function ___ehhandler$_ZN4core9panicking15panic_no_unwind17h799a74cf8a492a49E
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.64.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.96.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.73.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.18.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.41.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.98.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.84.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.117.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.93.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.69.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.102.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.37.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.74.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.60.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.38.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.20.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.48.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.8.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.10.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.24.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.124.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.81.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.27.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.111.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.30.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.36.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.23.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.32.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.95.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.59.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.21.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.50.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.65.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.16.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.72.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.91.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.89.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.55.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.47.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.119.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.116.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.88.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.85.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.83.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.44.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.7.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.28.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.45.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.1.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.6.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.71.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.63.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.43.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.31.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.90.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.12.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.101.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcore-5048a7bc95ecbe8e.rlib(core-5048a7bc95ecbe8e.core.8a217689-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.70.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.51.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          libcompiler_builtins-fe791603c7d507a8.rlib(compiler_builtins-fe791603c7d507a8.compiler_builtins.537a95b7-cgu.22.rcgu.o) : error LNK2001: unresolved external symbol __fltused
          LINK : error LNK2001: unresolved external symbol __DllMainCRTStartup@12
          LINK : error LNK2001: unresolved external symbol __load_config_used
          D:\a\rust\rust\build\i686-pc-windows-msvc\test\rustdoc\intra-doc\extern-lang-item-impl\auxiliary\extern_lang_item_impl_dep.dll : fatal error LNK1120: 9 unresolved externals

error: aborting due to previous error


---
Some tests failed in compiletest suite=rustdoc mode=rustdoc host=i686-pc-windows-msvc target=i686-pc-windows-msvc
test result: FAILED. 492 passed; 1 failed; 8 ignored; 0 measured; 0 filtered out; finished in 74.90s

Build completed unsuccessfully in 0:37:37
make: *** [Makefile:72: ci-subset-1] Error 1
```

</details>

--- a/compiler/rustc_metadata/src/rmeta/decoder.rs
+++ b/compiler/rustc_metadata/src/rmeta/decoder.rs
@@ -1032,13 +1032,11 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
     }
 
     /// Iterates over the language items in the given crate.
-    fn get_lang_items(self, tcx: TyCtxt<'tcx>) -> &'tcx [(DefId, usize)] {
-        tcx.arena.alloc_from_iter(
-            self.root
-                .lang_items
-                .decode(self)
-                .map(|(def_index, index)| (self.local_def_id(def_index), index)),
-        )
+    fn get_lang_items(self) -> impl Iterator<Item = (DefId, usize)> + 'a {
+        self.root
+            .lang_items
+            .decode(self)
+            .map(move |(def_index, index)| (self.local_def_id(def_index), index))
     }
 
     /// Iterates over the diagnostic items in the given crate.

--- a/compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs
+++ b/compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs
@@ -200,7 +200,7 @@ provide! { <'tcx> tcx, def_id, other, cdata,
         tcx.arena.alloc_slice(&result)
     }
     defined_lib_features => { cdata.get_lib_features(tcx) }
-    defined_lang_items => { cdata.get_lang_items(tcx) }
+    defined_lang_items => { tcx.arena.alloc_from_iter(cdata.get_lang_items()) }
     diagnostic_items => { cdata.get_diagnostic_items() }
     missing_lang_items => { cdata.get_missing_lang_items(tcx) }
 
@@ -500,6 +500,11 @@ impl CStore {
         cnum: CrateNum,
     ) -> impl Iterator<Item = (DefId, DefId)> + '_ {
         self.get_crate_data(cnum).get_inherent_impls()
+    }
+
+    /// Decodes all lang items in the crate (for rustdoc).
+    pub fn lang_items_untracked(&self, cnum: CrateNum) -> impl Iterator<Item = DefId> + '_ {
+        self.get_crate_data(cnum).get_lang_items().map(|(def_id, _)| def_id)
     }
 }
 

--- a/src/librustdoc/passes/collect_intra_doc_links/early.rs
+++ b/src/librustdoc/passes/collect_intra_doc_links/early.rs
@@ -118,6 +118,7 @@ impl IntraLinkCrateLoader<'_, '_> {
                 Vec::from_iter(self.resolver.cstore().trait_impls_in_crate_untracked(cnum));
             let all_inherent_impls =
                 Vec::from_iter(self.resolver.cstore().inherent_impls_in_crate_untracked(cnum));
+            let all_lang_items = Vec::from_iter(self.resolver.cstore().lang_items_untracked(cnum));
 
             // Querying traits in scope is expensive so we try to prune the impl and traits lists
             // using privacy, private traits and impls from other crates are never documented in
@@ -140,6 +141,9 @@ impl IntraLinkCrateLoader<'_, '_> {
                 if self.resolver.cstore().visibility_untracked(ty_def_id) == Visibility::Public {
                     self.add_traits_in_parent_scope(impl_def_id);
                 }
+            }
+            for def_id in all_lang_items {
+                self.add_traits_in_parent_scope(def_id);
             }
 
             self.all_traits.extend(all_traits);

--- a/src/test/rustdoc/intra-doc/auxiliary/extern-lang-item-impl-dep.rs
+++ b/src/test/rustdoc/intra-doc/auxiliary/extern-lang-item-impl-dep.rs
@@ -1,0 +1,29 @@
+// no-prefer-dynamic
+
+#![feature(lang_items)]
+
+#![crate_type = "rlib"]
+#![no_std]
+
+pub struct DerefsToF64(f64);
+
+impl core::ops::Deref for DerefsToF64 {
+    type Target = f64;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+mod inner {
+    #[lang = "f64_runtime"]
+    impl f64 {
+        /// [f64::clone]
+        pub fn method() {}
+    }
+}
+
+#[lang = "eh_personality"]
+fn foo() {}
+
+#[panic_handler]
+fn bar(_: &core::panic::PanicInfo) -> ! { loop {} }

--- a/src/test/rustdoc/intra-doc/extern-lang-item-impl.rs
+++ b/src/test/rustdoc/intra-doc/extern-lang-item-impl.rs
@@ -1,0 +1,11 @@
+// Reexport of a structure that derefs to a type with lang item impls having doc links in their
+// comments. The doc link points to an associated item, so we check that traits in scope for that
+// link are populated.
+
+// aux-build:extern-lang-item-impl-dep.rs
+
+#![no_std]
+
+extern crate extern_lang_item_impl_dep;
+
+pub use extern_lang_item_impl_dep::DerefsToF64;

